pkg/volume/iscsi/iscsi_util.go

```diff
@@ -133,16 +188,28 @@ func (util *ISCSIUtil) AttachDisk(b iscsiDiskMounter) error {
 		}
 		exist := waitForPathToExist(devicePath, 1, iscsiTransport)
 		if exist == false {
-			// discover iscsi target
-			out, err := b.plugin.execCommand("iscsiadm", []string{"-m", "discovery", "-t", "sendtargets", "-p", tp, "-I", b.iface})
+			// build discoverydb and discover iscsi target
+			b.plugin.execCommand("iscsiadm", []string{"-m", "discoverydb", "-t", "sendtargets", "-p", tp, "-I", b.iface, "-o", "new"})
+			// update discoverydb with CHAP secret
+			err = updateISCSIDiscoverydb(b, tp)
+			if err != nil {
+				last_err = fmt.Errorf("iscsi: failed to update discoverydb to portal %s error: %v", tp, err)
```

Comment:
IMO you should log the errors here (and everywhere where `last_err` is used), maybe as warnings or info, so we can see all the errors and not only the last one in the logs.